src/sonic-p4rt/Makefile

```diff
@@ -0,0 +1,47 @@
+.ONESHELL:
+SHELL = /bin/bash
+.SHELLFLAGS += -e
+
+BAZEL_TARGET_PATH = p4rt_app
+BAZEL_BUILD_TARGETS = //p4rt_app:p4rt_deb //p4rt_app:p4rt_dbg_deb
+
+# Enable debug symbols for remote debugging (generate a .dwp).
+BAZEL_BUILD_OPTS += --fission=yes --features=per_object_debug_info
+
+# Override sonic_swss_common to use the version specific to this buildimage.
+BAZEL_BUILD_OPTS += --override_repository=sonic_swss_common=/sonic/src/sonic-swss-common
+
+# Build optimized, stripped binaries.
+BAZEL_BUILD_OPTS += -c opt
+
+# Use a host directory to store Bazel cache, if mounted. This will speed up
+# incremental rebuilds on the same system for developers that makes changes
+# to sonic-pins code.
+#
+# To build with Bazel cache, add DOCKER_BUILDER_USER_MOUNT:
+#   DOCKER_BUILDER_USER_MOUNT=<directory on host system>:$BAZEL_CACHE:rw
+#
+# For example:
+#   DOCKER_BUILDER_USER_MOUNT=/tmp/bazel_cache:/bazel:rw make target/...
+#
+BAZEL_CACHE ?= /bazel
+BAZEL_OPTS += $(shell test -d $(BAZEL_CACHE) && echo --output_user_root=$(BAZEL_CACHE)/cache)
+
+MAIN_TARGET = $(SONIC_P4RT)
+DERIVED_TARGETS = $(SONIC_P4RT_DBG)
+
+$(addprefix $(DEST)/, $(MAIN_TARGET)): $(DEST)/% :
+	function cleanup {
+		# Note: make seems to hang if Bazel is still running
+		cd $(CURDIR)/sonic-pins && bazel $(BAZEL_OPTS) shutdown
+	}
+	trap cleanup EXIT
+	pushd ./sonic-pins
+	bazel $(BAZEL_OPTS) build $(BAZEL_BUILD_OPTS) $(BAZEL_BUILD_TARGETS)
+	bazel $(BAZEL_OPTS) test $(BAZEL_BUILD_OPTS) //$(BAZEL_TARGET_PATH)/...
+	pushd ./bazel-bin/$(BAZEL_TARGET_PATH)
```

Comment:
>pushd [](http://example.com/codeflow?start=1&length=5)

No need to pushd/popd just for mv?

Reply:
Sure
```suggestion
	mv ./bazel-bin/$(BAZEL_TARGET_PATH)
```

Reply:
The pushd makes things simpler here because Bazel won't build everything based on the call site (i.e. the packages are built under the bazel-bin/ directory). Which means all the $(DERIVED_TARGETS) would have to be updated for this one call.

Reply:
Leaving the pushd for the bazel build.

Then changing to the bazel-bin path (with cd) to keep the mv command short/clean. The first and only popd will return the original directory.